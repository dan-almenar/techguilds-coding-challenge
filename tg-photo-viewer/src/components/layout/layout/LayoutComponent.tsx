import styles from './layout.module.css'
import Navbar from '../navbar/Navbar.tsx';
import Footer from '../footer/Footer.tsx';

export default function LayoutComponent({
	children,
}: {
	children: React.ReactNode
}) {
	
  return (
	  <>
		<Navbar />
    	<main className={styles.main}>
			<h1>Hello, World!</h1>
			&nbsp;
			{ children }
			<span className={`material-icons-outlined ${styles.heart}`}>favorite_border</span>
    	</main>
		<Footer />
	  </>
  )
}
