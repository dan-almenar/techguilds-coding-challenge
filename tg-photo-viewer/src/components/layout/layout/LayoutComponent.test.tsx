// LayoutComponent integration test
import { render, screen } from '@testing-library/react';
import LayoutComponent from './LayoutComponent.tsx';

// mocked child component
const mockData = jest.mock('', () => {
	return {
		__esModule: true,
		default: jest.fn(() => {
			<>
				<section>
					<h2>Mocked Children</h2>
					<p>Some random paragraph</p>
				</section>
			</>
		}),
	};
}) as jest.Mocked<any>;

const { layout } = render(
	<LayoutComponent>
		{ mockData.default }
	</LayoutComponent>
)
// Tests
describe('LayoutComponent Tests', () => {
	render(
		<LayoutComponent>
			{ mockData.default }
		</LayoutComponent>
	)

	it('Mock children render properly', () => {
		expect({ layout }).toMatchSnapshot();
	});

	it('Layout component renders Navbar component', () => {
		render(
			<LayoutComponent>
				<div></div>
			</LayoutComponent>
		)

		const navbar = screen.getByRole('navigation');
		expect(navbar).toBeInTheDocument();
	});

	it('Layout component renders Footer component', () => {
		render(
			<LayoutComponent>
				<div></div>
			</LayoutComponent>
		)

		const footer = screen.getByRole('contentinfo');
		expect(footer).toBeInTheDocument();
	});
});

