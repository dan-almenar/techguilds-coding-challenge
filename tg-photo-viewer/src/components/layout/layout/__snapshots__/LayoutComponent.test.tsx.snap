// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutComponent Tests Mock children render properly 1`] = `
{
  "layout": undefined,
}
`;
